dashboard/app/models/concerns/pd/form.rb

```diff
@@ -34,6 +34,14 @@ def camelize_fields(fields)
     end
   end
 
+  # Determine if this (unsaved) model is a duplicate of an existing saved form.
+  # Used for idempotence check by controller
+  # @return [nil|Form] existing form that this is a duplicate of, or nil.
+  def find_existing
```

Comment:
I don't really like this name - `find_existing` doesn't find existing forms for other kinds of applications. You could just call it `idempotency_check`